### ISSUE TITLE
Support "route refresh"

### ISSUE THREAD

**Issue by ianthetechie** [stadiamaps/ferrostar#44]
Optionally, we should support configurable refresh behavior, whereby the SDK will periodically check that the current route is still the optimal one. This is useful in areas subject to traffic jams when combined with a routing solution that is aware of live traffic.

**Comment by Kandarp05** [comment]
Hey should we support configurable refresh behavior? This would let the SDK periodically check if the current route is still optimal, which could be useful in traffic-heavy areas. 
Thinking of adding a configurable route refresh strategy: Periodic (fixed interval), TrafficAware (considers traffic/incidents), and Custom (user-defined logic).
Should i go ahead and implement this?

**Comment by ianthetechie** [comment]
> Hey should we support configurable refresh behavior? This would let the SDK periodically check if the current route is still optimal, which could be useful in traffic-heavy areas.

Yeah, I think it should ultimately be configurable. The approach we usually take is to add some pretty easily configurable default cases, like "periodic" in this case. But provide hooks for custom implementations in the platform layer (Swift/Kotlin mainly) so that end users don't need to rely on us to merge a PR, or maintain a fork.

Traffic/incident aware is an interesting one, because don't actually have any one on the core dev team who has that as a core part of their use case. I think that will eventually be something we want to build out, but we can punt on that for now and just think about periodic, custom, and none (default) for the first pass to keep it simple.

> Should i go ahead and implement this?

Would you be up for a quick call to talk through it first (can arrange the details over a Slack DM; details on how to join are here: https://stadiamaps.github.io/ferrostar/contributing.html)? I think this is something a new contributor can tackle, but just want to make sure we're aligned real quick to save back + forth!
